### PR TITLE
Attempt to reduce test build cost

### PR DESCRIPTION
**Pull request by davidwrighton**
An evaluation of the cost of "Copy native test components to test output folder" indicates that there are 2 major performance bottlenecks in that routine.

There are the findings I had
1. The routine spends significant time loading every test project into msbuild to determine what to do. This processing takes as much as 400ms per test on hardware which has slow IO performance.
2. On Unix platforms, when producing the test script binaries, we make the scripts executable by using chmod. This ALSO takes as long as 400ms on the slowest platforms.
3. Actually copying the native binaries around is extremely cheap and fast, and the cost to generate the test scripts is also fairly low.

This change addresses both of these issues.
1. For the issue that we look into each test project to determine what to do, I observed that for tests which are pri1 tests, when running a pri0 only job, we don't need to even examine any of the tests marked as exclusively pri1. This is a bit of a hack, but it does seem to work quite well.
2. For the chmod issue, since we don't actually maintain the executable bit when sent to helix, I set an msbuild property when generating the scripts during the "Copy native test components to test output folder" to just not set that flag.

Results:
Mac OSX test run jobs take ~8 minutes to run this portion of the build, down from ~27 minutes. Performance improvements are visible on all architectures/OS pairings, but they aren't as dramatic.

### CODE REVIEW

File: src/coreclr/vm/zapsig.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Licensed to the .NET Foundation under one or more agreements.
+ // Licensed to the .NET Foundation under one or more agreements.
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/vm/zapsig.cpp:1]:
```suggestion
// Licensed to the .NET Foundation under one or more agreements.
```

Comment (davidwrighton) [reply src/coreclr/vm/zapsig.cpp]: Yeah, that change was only in there to make sure the build would trigger at least the coreclr stuff to run. As it appears it triggers everything anyways, I've reverted it.

**Comment by davidwrighton** [comment]
@jkoritzinsky I think this is still running the same set of tests, but I don't actually know how to validate that. Do you?

**Comment by jkoritzinsky** [comment]
I don't think I have a script for verifying that today. @trylek might have one.

I know that Helix is a little unreliable for recording successful tests in Kusto, but otherwise that might be a good place to look.

**Comment by hoyosjs** [comment]
My best guess is the AzDO API for the test run, since that's where Helix reports to and then there's back ingestion into Kusto

File: src/tests/Common/dirs.proj [hunk 2]
```diff
@@ -45,11 +45,17 @@
...
-
     <!-- All the test projects are partitioned into the test groups as defined below.
          Each of the test groups is meant to be built by a separate MSBuild invocation with specified $(__TestGroupToBuild) property. -->
-
     <ItemGroup Condition=" '$(CLRTestPriorityToBuild)' == '0' ">
+
+      <!-- Fast filter to avoid files with Pri1 projects, this isn't completely accurate, but its close. Any Pri1 tests not caught by this will be skipped,
+           and any tests intended as Pri0 will end up running as Pri1, which isn't terrible. This trick allows us to avoid loading each project into msbuild
+           and evaluating it, which is a substantial improvement to the performance of the "Copy native test components to test output folder" task in our test runs  -->
+      <Pri1Projects Include="@(AllProjects)" Condition="$([System.Text.RegularExpressions.Regex]::IsMatch($([System.IO.File]::ReadAllText('%(AllProjects.FullPath)')), '&lt;CLRTestPriority&gt;1&lt;/CLRTestPriority&gt;'))" />
```
Comment (trylek) [inline src/tests/Common/dirs.proj:55]: Beautiful!

**Review by trylek — APPROVED** [review]
Looks great to me, thanks David! I suspected there may be some low hanging fruits. One suggestion down on my quality week list :-).

**Comment by trylek** [comment]
I don't have any targeted script for evaluating the set of tests run. AzDO UI does report the total number of tests so comparing it to neighboring runs without this change should give us a pretty decent indication whether this is basically fine or whether there's some big problem lurking somewhere.

**Comment by trylek** [comment]
When merging the JIT/Methodical tests, I used to compare the XUnit xml outputs from the non-merged and the merged runs until I hit a match but I no longer remember whether I had any automation for this.

**Comment by davidwrighton** [comment]
I've validated that AzDO thinks we run the same number of tests.

**Pull request merged**